internal/providers/terraform/aws/kinesis_firehose_delivery_stream.go

```diff
@@ -0,0 +1,131 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetKenesisFirehoseDeliveryStreamRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_kinesis_firehose_delivery_stream",
+		RFunc: NewKenesisFirehoseDeliveryStream,
+	}
+}
+
+func NewKenesisFirehoseDeliveryStream(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	region := d.Get("region").String()
+	costComponents := make([]*schema.CostComponent, 0)
+	var monthlyDataIngestedGb *decimal.Decimal
+	var requestQuantities []decimal.Decimal
+	halfAMillion := []int{512000}
+	oneAndHalfPB := decimal.NewFromInt(1536000)
+
+	if u != nil && u.Get("monthly_data_ingested_gb").Type != gjson.Null {
+		monthlyDataIngestedGb = decimalPtr(decimal.NewFromInt(u.Get("monthly_data_ingested_gb").Int()))
+		requestQuantities = usage.CalculateTierBuckets(*monthlyDataIngestedGb, halfAMillion)
+		firstGB := requestQuantities[0]
+		nextOneAndHalfGB := requestQuantities[1]
+
+		if nextOneAndHalfGB.Equal(decimal.Zero) {
+
+			costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", &firstGB))
+		}
+		if nextOneAndHalfGB.GreaterThan(oneAndHalfPB) {
+			nextThreeGB := nextOneAndHalfGB.Sub(oneAndHalfPB)
+			costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", decimalPtr(decimal.NewFromInt(int64(halfAMillion[0])))))
+			costComponents = append(costComponents, kenesisFirehosCostComponent("next 1.5PB", region, "512000", "2048000", &oneAndHalfPB))
+			costComponents = append(costComponents, kenesisFirehosCostComponent("next 3PB", region, "2048000", "Inf", &nextThreeGB))
+		} else {
+			costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", decimalPtr(decimal.NewFromInt(int64(halfAMillion[0])))))
+			costComponents = append(costComponents, kenesisFirehosCostComponent("next 1.5PB", region, "512000", "2048000", &nextOneAndHalfGB))
+		}
+	} else {
+		var unknown *decimal.Decimal
+
+		costComponents = append(costComponents, kenesisFirehosCostComponent("first 500TB", region, "0", "512000", unknown))
+		costComponents = append(costComponents, kenesisFirehosCostComponent("next 1.5PB", region, "512000", "2048000", unknown))
+		costComponents = append(costComponents, kenesisFirehosCostComponent("next 3PB", region, "2048000", "Inf", unknown))
+	}
+
+	if d.Get("extended_s3_configuration.0.data_format_conversion_configuration.0.enabled").Type != gjson.Null {
+		enabled := d.Get("extended_s3_configuration.0.data_format_conversion_configuration.0.enabled")
+		if enabled.Type == gjson.True {
+			costComponents = append(costComponents, kenesisFirehosConversationCostComponent(region, monthlyDataIngestedGb))
+		}
+	}
```

Comment:
I think you can compare d.Get(..).Type with gjson.True, as it doesn't have too much sense to doing double type comparison
```suggestion
	if d.Get("extended_s3_configuration.0.data_format_conversion_configuration.0.enabled").Type == gjson.True {
		costComponents = append(costComponents, kenesisFirehosConversationCostComponent(region, monthlyDataIngestedGb))
	}
```